### ISSUE TITLE
Support variables param in createPipeline

### ISSUE THREAD

**Issue by SamerAdra** [gmessner/gitlab4j-api#321]
Add support for the optional "variables" parameter in the createPipeline call.
Would be good if this could be implemented as an overload of createPipeline accepting the following parameters: Object projectIdOrPath, String ref, Map<String, String> variables

See https://docs.gitlab.com/ee/api/pipelines.html#create-a-new-pipeline

**Comment by gmessner** [comment]
@SamerAdra 
The documentation is very unclear on what the URL or form data should look like when variables are included, can you help by getting that information?

**Comment by gmessner** [comment]
@SamerAdra 
I located the info I needed for this at https://docs.gitlab.com/ce/api/#array-of-hashes

This has been implemented and will be released in the next few days.

**Comment by gmessner** [comment]
@SamerAdra 
Release 4.9.19 with this functionality has been released. Please verify and close when you have a chance.

**Comment by gmessner** [comment]
@SamerAdra 
Since the enhancement has been released I am closing this issue, feel free to open it back up if you find any problems with the implementation.

**Comment by SamerAdra** [comment]
Sorry for waiting so long to test it. But I found out that it's not working when calling createPipeline with variables.
GitLab's API documentation on this does not help, but I found a comment in an [issue](https://gitlab.com/gitlab-org/gitlab-ce/issues/49527) that seems to help. Rémy Coutable suggests passing in the ref and variables as URL parameters, even though it's a POST call. I tried it out and it does seem to work that way.

org.gitlab4j.api.GitLabApiException: The following fields have validation errors: variables.key, base

	at org.gitlab4j.api.AbstractApi.validate(AbstractApi.java:593)
	at org.gitlab4j.api.AbstractApi.post(AbstractApi.java:319)
	at org.gitlab4j.api.PipelineApi.createPipeline(PipelineApi.java:254)
